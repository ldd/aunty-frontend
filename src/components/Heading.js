import React from "react";

const Heading = ({ title, subtitle }) => (
  <>
    {title && <h1 className="title">{title}</h1>}
    {subtitle && <h2 className="subtitle">{subtitle}</h2>}
  </>
);

export default Heading;
