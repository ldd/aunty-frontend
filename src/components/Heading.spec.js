import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Heading from "./Heading";

describe("Heading", () => {
  it("renders snapshot [empty]", () => {
    const component = renderer.create(<Heading />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders snapshot [title && subtitle]", () => {
    const title = "someTitle";
    const subtitle = "someSubtitle";
    const component = renderer.create(
      <Heading title={title} subtitle={subtitle} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render no title nor subtitle", () => {
    const wrapper = shallow(<Heading />);
    expect(wrapper.equals(<></>)).toBe(true);
  });
  it("should render a title and subtitle", () => {
    const title = "someTitle";
    const wrapper = shallow(<Heading title={title} />);
    expect(wrapper.someWhere(node => node.text() === title)).toBe(true);
  });
  it("should render a title and subtitle", () => {
    const title = "someTitle";
    const subtitle = "someSubtitle";
    const wrapper = shallow(<Heading title={title} subtitle={subtitle} />);
    expect(
      wrapper.find(".title").someWhere(node => node.text() === title)
    ).toBe(true);
    expect(
      wrapper.find(".subtitle").someWhere(node => node.text() === subtitle)
    ).toBe(true);
  });
});
