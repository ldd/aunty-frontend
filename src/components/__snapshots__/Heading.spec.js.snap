// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders snapshot [empty] 1`] = `null`;

exports[`Heading renders snapshot [title && subtitle] 1`] = `
Array [
  <h1
    className="title"
  >
    someTitle
  </h1>,
  <h2
    className="subtitle"
  >
    someSubtitle
  </h2>,
]
`;
